### ISSUE TITLE
Slotted mixins cause an error

### ISSUE THREAD

**Issue by Rapptz** [microsoft/pyright#2039]
**Describe the bug**
Thanks to #2022 Pyright gained the ability to check for slots. However there's an edge case I found that wasn't accounted for involving mixins. Essentially, when using a mixin and assigning an attribute to it an error is reported. In Python in order for the mixins to play nicely they must be defined with `__slots__ = ()`.

**To Reproduce**
The following code fails a type check:

```py
class Mixin:
    __slots__ = ()
    value: int

    def foo(self, value: int) -> None:
        self.value = value * 20

class B(Mixin):
    def __init__(self):
        self.value = 30

b = B()
b.foo(60)
```

**Expected behavior**
This should type check.

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/1695103/123728151-d9f36880-d860-11eb-8279-ac7d3c0a8f2c.png)

**VS Code extension or command-line**
Pyright 1.1.153

**Additional context**
This usage of empty slots is necessary for a mixin to work well with slotted classes. While one would expect to just be able to assign the slots as needed in the mixin class, this causes issues with multiple inheritance by raising a `TypeError`: 

```py
class MixinA:
    __slots__ = ('a',)

class MixinB:
    __slots__ = ('b',)

class C(MixinA, MixinB):
    __slots__ = ('c',)
```
Would raise with `TypeError: multiple bases have instance lay-out conflict`. This is mentioned in the [documentation](https://docs.python.org/3/reference/datamodel.html?#notes-on-using-slots) as well:

> Multiple inheritance with multiple slotted parent classes can be used, but only one parent is allowed to have attributes created by slots (the other bases must have empty slot layouts) - violations raise TypeError.

I think the best solution would be to treat `__slots__ = ()` as if no slots are defined at all since it's a bit of a special case. Alternatively if this is undesired, to special case the mixin behaviour then a class attribute type hint could be used to signal that the attribute will be around later (like `value: int` here).

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.154, which I just published. It will also be included in the next release of pylance.
